google/cloud/spanner/value.h

```diff
@@ -135,6 +132,23 @@ class Value {
     return ProtoEqual(type_, GetType(T{}));
   }
 
+  /**
+   * Returns true if is<T>() and the contained value is "null".
+   *
+   * Example:
+   *
+   *     spanner::Value v{true};
+   *     assert(!v.is_null<bool>());
+   *
+   *     spanner::Value null_v = spanner::Value::MakeNull<bool>();
+   *     assert(v.is_null<bool>());
+   *     assert(!v.is_null<std::int64_t>());
+   */
+  template <typename T>
+  bool is_null() const {
+    return is<T>() && value_.kind_case() == google::protobuf::Value::kNullValue;
```

Comment:
Consider instead `return !HasValue<T>(type_, value_).ok();`

Reply:
That looks different.  In particular, ...
```
Value null_a = spanner::Value::MakeNull<A>();
EXPECT_FALSE(null_a.is<B>());
```
would fail.  I added a test for that case.

Reply:
Ahh, right.